app/models/favorite_project.rb

```diff
@@ -1,4 +1,4 @@
 class FavoriteProject < ApplicationRecord
   belongs_to :project
-  belongs_to :user
+  belongs_to :user, counter_cache: true # count for analytics
```

Comment:
nit: Not sure you need the comment. We can actually use this counters for more things... 
But up to you

Reply:
Updated

Reply:
# use .size for cache, use .count to force COUNT query